vault/resource_oci_auth_backend_config.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func ociAuthBackendConfigResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ociAuthBackendWrite,
+		ReadContext:   ReadContextWrapper(ociAuthBackendRead),
+		UpdateContext: ociAuthBackendWrite,
+		DeleteContext: ociAuthBackendDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "oci",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"home_tenancy_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: " The Tenancy OCID of your OCI account.",
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func ociAuthBackendWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	// if backend comes from the config, it won't have the StateFunc
+	// applied yet, so we need to apply it again.
+	backend := d.Get("backend").(string)
+	homeTenancyId := d.Get("home_tenancy_id").(string)
+
+	path := ociAuthBackendConfigPath(backend)
+
+	data := map[string]interface{}{
+		"home_tenancy_id": homeTenancyId,
+	}
+
+	log.Printf("[DEBUG] Writing OCI auth backend config to %q", path)
+	_, err := config.Logical().Write(path, data)
+	if err != nil {
```

Comment:
There is also `client.Sys().EnableAuthWithOptionsWithContext()` available for use here which has context and handles the path formatting automatically, so you can just pass in the `backend` (or path, see above comment) directly.